jina/proto/jina.proto

```diff
@@ -104,6 +104,9 @@ message Document {
     NdArray embedding = 19;
 
     Relevance score = 20;
+
+    // mode_id, an identifier to the modality this document belongs to. In the scope of multi/cross modal search
+    string mode_id = 21;
```

Comment:
For the sake of completion, should it be a repeated field? so that a document can have more than one mode and then it can be split into chunks of different modes?

Reply:
In addition to this, in content should we consider another field option that is any repeated combination of content? or should we consider that if we use multimodality, one should pass content that is his/her responsibility to split (he/she should provide a crafter able to split into different modes if required)?

Reply:
Any thoughts on this @hanxiao, @nan-wang

Reply:
let me switch to this branch

Reply:
if it is called `something_id` then better keep it as `int`.

maybe `repeated string modalities`?

in principal it can be also part of `tags`, but given this is one of the highlights of Jina, we promote this field as an independent field.

> For the sake of completion, should it be a repeated field? so that a document can have more than one mode and then it can be split into chunks of different modes?

Reply:
I would agree with ` repeated string modalities`. I think a string can have more meaningful names that can map business or function value. Another point is whether we need to keep multiple content types in the same document?.

Reply:
do you mean something like this?
```protobuf
message MixedContent {
  repeated Content values = 1;
}

message Content {
  oneof content {
    bytes buffer = 1;
    
    NdArray blob = 2;
    
    string text = 3;
    
    MixedContent mix = 4;
  }
}
```

and then 

```protobuf
message Document {
   //...
   Content content = 1
   //...
}
```

Reply:
yes, something like this. I would expect that if I reconstruct the content of my chunks, I can exactly map it into the content of the parent document. Does it make sense? Or is it adding another layer of complexity?

Reply:
let's keep the content "flat" as it is and uses nested "Chunks" to improve the expressivity of a Document.

Reply:
Ok, but then if someone wants to build a `crafter` to split them, it must have the content in a way that he/she can split, right? But I agree, let's keep it for now `flat` and if we see that is needed we can always add later

Reply:
Imagine when a doc first enters Jina Flow, it is a quite reasonable and general assumption that this doc is always "single-content" as `buffer.

It then depends on the `Crafter`/`Segmenter` to interpret the buffer and translate it into more readable content. The multi-modality interpretation happens inside the Flow.

If we assume a doc enters Jina with "multi-content", then this means there is a powerful client does what our `Crafter` supposed to do: it reads from buffer and parses it and fills in the "multi-content".

We can not make such assumption on the client side, because:
- it significantly increases the maintaining cost of clients (especially when it comes to multi-language)
- it is easier to scale and maintain when such crafting job is done inside the flow.

Reply:
now i start to think we should use `string modality` rather than `repeated string`

> let's keep the content "flat" as it is and uses nested "Chunks" to improve the expressivity of a Document.

Reply:
given that a recursive document can give us *infinite* expressivity, we should really keep the fields inside `Document` simple and flat